docs/attributes.md

```diff
@@ -66,3 +66,69 @@ _Example_: Extract `src` and `alt` from each image element in the block's markup
 //   ]
 // }
 ```
+
+## Meta
+
+Attributes may be obtained from a post's meta rather than from the block's representation in saved post content. For this, an attribute is required to specify its corresponding meta key under the `meta` key:
+
+```js
+attributes: {
+	author: {
+		type: 'string',
+		meta: 'author'
+	},
+},
+```
+
+From here, meta attributes can be read and written by a block using the same interface as any attribute:
+
+{% codetabs %}
+{% ES5 %}
+```js
+edit: function( props ) {
+	function onChange( event ) {
+		props.setAttributes( { author: event.target.value } );
+	}
+
+	return el( 'input', {
+		value: props.attributes.author,
+		onChange: onChange,
+	} );
+},
+```
+{% ESNext %}
+```js
+edit( { attributes, setAttributes } ) {
+	function onChange( event ) {
+		setAttributes( { author: event.target.value } );
+	}
+
+	return <input value={ attributes.author } onChange={ onChange } />;
+},
+```
+{% end %}
+
+### Considerations
+
+By default, a meta field will be excluded from a post object's meta. This can be circumvented by explicitly making the field visible:
```

Comment:
@pento this would be nice to address in general :)

Reply:
Yeah, that's annoying.

@rmccue: I don't recall the history for why postmeta is only included in a response when it has the `show_in_rest` flag set. (As opposed to at least returning the visible postmeta when the user is appropriately authenticated.)

Reply:
Meta fields aren't guaranteed to be "safe" by default. "Safe" is a few things, namely safe for JSON serialisation (which is lossy compared to PHP), and safe for capabilities (not all meta is always available).

The current behaviour exists for those reasons. When we initially didn't require `show_in_rest`, people noted that this would expose hidden fields created by plugin or users (turns out, a bunch of people use the "Custom Fields" metabox for random internal notes). Solving these problems in a consistent and safe way turned out to be essentially impossible. 

`show_in_rest` is an indicator that plugin developers are accepting the limitations of the API for their meta fields.

Reply:
These hidden fields would be shown if you do `get_post_meta()` calls, no? It feels like the API should adhere to the expectations of doing such authenticated calls.